src/java/org/apache/nutch/crawl/Injector.java

```diff
@@ -402,40 +402,38 @@ public void inject(Path crawlDb, Path urlDir, boolean overwrite,
 
     // lock an existing crawldb to prevent multiple simultaneous updates
     Path lock = CrawlDb.lock(conf, crawlDb, false);
```

Comment:
See the discussion in NUTH-3078: moving the creating of the lockfile down just before the job is run would avoid that a failure during job configuration leads to a lost lockfile.

Reply:
Great!